pkg/apply/taskrunner/task.go

```diff
@@ -243,8 +277,9 @@ func (w *WaitTask) StatusUpdate(taskContext *TaskContext, id object.ObjMetadata)
 		}
 	}
 
-	// if all conditions are met, complete the wait task
-	if conditionMet(taskContext, w.pending, w.Condition) {
+	// If we no longer have any pending resources, the WaitTask
+	// can be completed.
+	if len(w.pending) == 0 {
 		// all reconciled - clear pending and exit
 		w.pending = object.ObjMetadataSet{}
```

Comment:
Since `len(w.pending) == 0`, we can remove this line.

Reply:
That is true. I have removed that line.